File: python-pr/52/python/code-quality-readability-maintainability/file_processor_31.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def process_text_file(filename):
    try:
        with open(filename, 'r') as file:
            lines = file.readlines()
        
        processed_lines = []
        for line in lines:
            cleaned_line = line.strip()
            if not cleaned_line:
                continue
                
            words = cleaned_line.split()
            significant_words = [word for word in words if len(word) > 3]
            
            if significant_words:
                processed_lines.append(' '.join(significant_words))
        
        return processed_lines
    
    except FileNotFoundError:
        return []
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/file_processor_31.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function processTextFile(filename) {
    const fs = require('fs');
    
    try {
        const data = fs.readFileSync(filename, 'utf8');
        const lines = data.split('\n');
        
        return lines
            .map(line => line.trim())
            .filter(line => line.length > 0)
            .map(line => {
                const words = line.split(' ');
                const significantWords = words.filter(word => word.length > 3);
                return significantWords.join(' ');
            })
            .filter(processedLine => processedLine.length > 0);
    } catch (error) {
        console.error('Error processing file:', error.message);
        return [];
    }
}
// {/fact}